### ISSUE TITLE
1.4.0 RC1+7: *** glibc detected *** python: corrupted double-linked list

### ISSUE THREAD

**Issue by letmaik** [matplotlib/matplotlib#3304]
I just spent a day figuring out where `*** glibc detected *** .../bin/python: corrupted double-linked list: 0x00000000069a01a0 ***` was coming from. I installed the latest 1.4.x branch version, so I thought that something must have happened there. Bisecting the commits and testing different versions, I found that  7b39e78b0807b8176183a5890b502f0f521b2465 introduced the problem. The errors happen randomly and the stacktraces are different most of the time. Some examples:

```
*** glibc detected *** .../p27/bin/python: corrupted double-linked list: 0x00000000069a01a0 ***
======= Backtrace: =========
/lib64/libc.so.6(+0x76166)[0x7f95f35c2166]
/lib64/libc.so.6(+0x79f9a)[0x7f95f35c5f9a]
/lib64/libc.so.6(__libc_malloc+0x71)[0x7f95f35c69a1]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyString_FromStringAndSize+0xda)[0x7f95f4224c6a]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyUnicodeUCS2_EncodeUTF8+0x1be)[0x7f95f424860e]
.../p27/lib/python2.7/lib-dynload/_io.so(+0x15111)[0x7f95ec053111]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalFrameEx+0x5910)[0x7f95f427f000]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalCodeEx+0x88e)[0x7f95f4280b7e]
/opt/Python2.7/lib/libpython2.7.so.1.0(+0x76f28)[0x7f95f41fef28]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyObject_Call+0x53)[0x7f95f41cf9a3]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalFrameEx+0x412a)[0x7f95f427d81a]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalCodeEx+0x88e)[0x7f95f4280b7e]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalFrameEx+0x563a)[0x7f95f427ed2a]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalCodeEx+0x88e)[0x7f95f4280b7e]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalFrameEx+0x563a)[0x7f95f427ed2a]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalCodeEx+0x88e)[0x7f95f4280b7e]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalFrameEx+0x563a)[0x7f95f427ed2a]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalCodeEx+0x88e)[0x7f95f4280b7e]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalFrameEx+0x563a)[0x7f95f427ed2a]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalCodeEx+0x88e)[0x7f95f4280b7e]
/opt/Python2.7/lib/libpython2.7.so.1.0(+0x76f28)[0x7f95f41fef28]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyObject_Call+0x53)[0x7f95f41cf9a3]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalFrameEx+0x412a)[0x7f95f427d81a]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalCodeEx+0x88e)[0x7f95f4280b7e]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalFrameEx+0x563a)[0x7f95f427ed2a]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalCodeEx+0x88e)[0x7f95f4280b7e]
/opt/Python2.7/lib/libpython2.7.so.1.0(+0x76f28)[0x7f95f41fef28]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyObject_Call+0x53)[0x7f95f41cf9a3]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalFrameEx+0x412a)[0x7f95f427d81a]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalCodeEx+0x88e)[0x7f95f4280b7e]
/opt/Python2.7/lib/libpython2.7.so.1.0(+0x76e21)[0x7f95f41fee21]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyObject_Call+0x53)[0x7f95f41cf9a3]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalFrameEx+0x412a)[0x7f95f427d81a]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalCodeEx+0x88e)[0x7f95f4280b7e]
/opt/Python2.7/lib/libpython2.7.so.1.0(+0x76f28)[0x7f95f41fef28]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyObject_Call+0x53)[0x7f95f41cf9a3]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalFrameEx+0x412a)[0x7f95f427d81a]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalCodeEx+0x88e)[0x7f95f4280b7e]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalFrameEx+0x563a)[0x7f95f427ed2a]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalCodeEx+0x88e)[0x7f95f4280b7e]
/opt/Python2.7/lib/libpython2.7.so.1.0(+0x76f28)[0x7f95f41fef28]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyObject_Call+0x53)[0x7f95f41cf9a3]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalFrameEx+0x412a)[0x7f95f427d81a]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalCodeEx+0x88e)[0x7f95f4280b7e]
/opt/Python2.7/lib/libpython2.7.so.1.0(+0x76f28)[0x7f95f41fef28]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyObject_Call+0x53)[0x7f95f41cf9a3]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalFrameEx+0x412a)[0x7f95f427d81a]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalCodeEx+0x88e)[0x7f95f4280b7e]
/opt/Python2.7/lib/libpython2.7.so.1.0(+0x76f28)[0x7f95f41fef28]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyObject_Call+0x53)[0x7f95f41cf9a3]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalFrameEx+0x412a)[0x7f95f427d81a]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalCodeEx+0x88e)[0x7f95f4280b7e]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalFrameEx+0x563a)[0x7f95f427ed2a]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalCodeEx+0x88e)[0x7f95f4280b7e]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalFrameEx+0x563a)[0x7f95f427ed2a]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalCodeEx+0x88e)[0x7f95f4280b7e]
/opt/Python2.7/lib/libpython2.7.so.1.0(+0x76f28)[0x7f95f41fef28]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyObject_Call+0x53)[0x7f95f41cf9a3]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalFrameEx+0x412a)[0x7f95f427d81a]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalCodeEx+0x88e)[0x7f95f4280b7e]
/opt/Python2.7/lib/libpython2.7.so.1.0(+0x76f28)[0x7f95f41fef28]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyObject_Call+0x53)[0x7f95f41cf9a3]
.../p27/lib/python2.7/lib-dynload/_functools.so(+0x1764)[0x7f95eba30764]
```

```
*** glibc detected *** .../p27/bin/python: corrupted double-linked list: 0x000000001771e600 ***
======= Backtrace: =========
/lib64/libc.so.6(+0x76166)[0x7ffd5f3be166]
/lib64/libc.so.6(+0x79f9a)[0x7ffd5f3c1f9a]
/lib64/libc.so.6(__libc_malloc+0x71)[0x7ffd5f3c29a1]
.../p27/lib/python2.7/site-packages/numpy/core/multiarray.so(+0xa5782)[0x7ffd52c8f782]
.../p27/lib/python2.7/site-packages/numpy/core/multiarray.so(+0xe5eb6)[0x7ffd52ccfeb6]
.../p27/lib/python2.7/site-packages/numpy/core/multiarray.so(+0xe64be)[0x7ffd52cd04be]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalFrameEx+0x5910)[0x7ffd6007b000]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalCodeEx+0x88e)[0x7ffd6007cb7e]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalFrameEx+0x563a)[0x7ffd6007ad2a]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalCodeEx+0x88e)[0x7ffd6007cb7e]
/opt/Python2.7/lib/libpython2.7.so.1.0(+0x76e21)[0x7ffd5fffae21]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyObject_Call+0x53)[0x7ffd5ffcb9a3]
/opt/Python2.7/lib/libpython2.7.so.1.0(+0x5a0cf)[0x7ffd5ffde0cf]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyObject_Call+0x53)[0x7ffd5ffcb9a3]
/opt/Python2.7/lib/libpython2.7.so.1.0(_PyObject_CallFunction_SizeT+0xac)[0x7ffd5ffcecbc]
.../p27/lib/python2.7/site-packages/numpy/core/multiarray.so(+0x61909)[0x7ffd52c4b909]
.../p27/lib/python2.7/site-packages/numpy/core/multiarray.so(+0x64a30)[0x7ffd52c4ea30]
.../p27/lib/python2.7/site-packages/numpy/core/multiarray.so(+0x65ab3)[0x7ffd52c4fab3]
.../p27/lib/python2.7/site-packages/matplotlib/_path.so(_Z31py_to_agg_transformation_matrixP7_objectb+0x74)[0x7ffd48ddb994]
.../p27/lib/python2.7/site-packages/matplotlib/_path.so(_ZN12_path_module14convert_to_svgERKN2Py5TupleE+0xb3)[0x7ffd48dc4003]
.../p27/lib/python2.7/site-packages/matplotlib/_path.so(_ZN2Py15ExtensionModuleI12_path_moduleE21invoke_method_varargsEPvRKNS_5TupleE+0x18)[0x7ffd48dcf828]
.../p27/lib/python2.7/site-packages/matplotlib/_path.so(method_varargs_call_handler+0x15d)[0x7ffd48dea20d]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalFrameEx+0x5910)[0x7ffd6007b000]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalCodeEx+0x88e)[0x7ffd6007cb7e]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalFrameEx+0x563a)[0x7ffd6007ad2a]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalCodeEx+0x88e)[0x7ffd6007cb7e]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalFrameEx+0x563a)[0x7ffd6007ad2a]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalCodeEx+0x88e)[0x7ffd6007cb7e]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalFrameEx+0x563a)[0x7ffd6007ad2a]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalCodeEx+0x88e)[0x7ffd6007cb7e]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalFrameEx+0x563a)[0x7ffd6007ad2a]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalCodeEx+0x88e)[0x7ffd6007cb7e]
/opt/Python2.7/lib/libpython2.7.so.1.0(+0x76f28)[0x7ffd5fffaf28]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyObject_Call+0x53)[0x7ffd5ffcb9a3]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalFrameEx+0x412a)[0x7ffd6007981a]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalCodeEx+0x88e)[0x7ffd6007cb7e]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalFrameEx+0x563a)[0x7ffd6007ad2a]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalCodeEx+0x88e)[0x7ffd6007cb7e]
/opt/Python2.7/lib/libpython2.7.so.1.0(+0x76f28)[0x7ffd5fffaf28]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyObject_Call+0x53)[0x7ffd5ffcb9a3]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalFrameEx+0x412a)[0x7ffd6007981a]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalCodeEx+0x88e)[0x7ffd6007cb7e]
/opt/Python2.7/lib/libpython2.7.so.1.0(+0x76e21)[0x7ffd5fffae21]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyObject_Call+0x53)[0x7ffd5ffcb9a3]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalFrameEx+0x412a)[0x7ffd6007981a]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalCodeEx+0x88e)[0x7ffd6007cb7e]
/opt/Python2.7/lib/libpython2.7.so.1.0(+0x76f28)[0x7ffd5fffaf28]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyObject_Call+0x53)[0x7ffd5ffcb9a3]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalFrameEx+0x412a)[0x7ffd6007981a]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalCodeEx+0x88e)[0x7ffd6007cb7e]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalFrameEx+0x563a)[0x7ffd6007ad2a]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalCodeEx+0x88e)[0x7ffd6007cb7e]
/opt/Python2.7/lib/libpython2.7.so.1.0(+0x76f28)[0x7ffd5fffaf28]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyObject_Call+0x53)[0x7ffd5ffcb9a3]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalFrameEx+0x412a)[0x7ffd6007981a]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalCodeEx+0x88e)[0x7ffd6007cb7e]
/opt/Python2.7/lib/libpython2.7.so.1.0(+0x76f28)[0x7ffd5fffaf28]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyObject_Call+0x53)[0x7ffd5ffcb9a3]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalFrameEx+0x412a)[0x7ffd6007981a]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalCodeEx+0x88e)[0x7ffd6007cb7e]
/opt/Python2.7/lib/libpython2.7.so.1.0(+0x76f28)[0x7ffd5fffaf28]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyObject_Call+0x53)[0x7ffd5ffcb9a3]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalFrameEx+0x412a)[0x7ffd6007981a]
```

```
*** glibc detected *** .../p27/bin/python: corrupted double-linked list: 0x0000000005905b40 ***
======= Backtrace: =========
/lib64/libc.so.6(+0x76166)[0x7ff01fe0c166]
/lib64/libc.so.6(+0x79f9a)[0x7ff01fe0ff9a]
/lib64/libc.so.6(__libc_malloc+0x71)[0x7ff01fe109a1]
.../p27/lib/python2.7/site-packages/matplotlib/_path.so(_ZN12_path_module14convert_to_svgERKN2Py5TupleE+0x92c)[0x7ff00981287c]
.../p27/lib/python2.7/site-packages/matplotlib/_path.so(_ZN2Py15ExtensionModuleI12_path_moduleE21invoke_method_varargsEPvRKNS_5TupleE+0x18)[0x7ff00981d828]
.../p27/lib/python2.7/site-packages/matplotlib/_path.so(method_varargs_call_handler+0x15d)[0x7ff00983820d]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalFrameEx+0x5910)[0x7ff020ac9000]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalCodeEx+0x88e)[0x7ff020acab7e]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalFrameEx+0x563a)[0x7ff020ac8d2a]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalCodeEx+0x88e)[0x7ff020acab7e]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalFrameEx+0x563a)[0x7ff020ac8d2a]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalCodeEx+0x88e)[0x7ff020acab7e]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalFrameEx+0x563a)[0x7ff020ac8d2a]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalCodeEx+0x88e)[0x7ff020acab7e]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalFrameEx+0x563a)[0x7ff020ac8d2a]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalCodeEx+0x88e)[0x7ff020acab7e]
/opt/Python2.7/lib/libpython2.7.so.1.0(+0x76f28)[0x7ff020a48f28]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyObject_Call+0x53)[0x7ff020a199a3]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalFrameEx+0x412a)[0x7ff020ac781a]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalCodeEx+0x88e)[0x7ff020acab7e]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalFrameEx+0x563a)[0x7ff020ac8d2a]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalCodeEx+0x88e)[0x7ff020acab7e]
/opt/Python2.7/lib/libpython2.7.so.1.0(+0x76f28)[0x7ff020a48f28]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyObject_Call+0x53)[0x7ff020a199a3]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalFrameEx+0x412a)[0x7ff020ac781a]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalCodeEx+0x88e)[0x7ff020acab7e]
/opt/Python2.7/lib/libpython2.7.so.1.0(+0x76e21)[0x7ff020a48e21]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyObject_Call+0x53)[0x7ff020a199a3]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalFrameEx+0x412a)[0x7ff020ac781a]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalCodeEx+0x88e)[0x7ff020acab7e]
/opt/Python2.7/lib/libpython2.7.so.1.0(+0x76f28)[0x7ff020a48f28]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyObject_Call+0x53)[0x7ff020a199a3]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalFrameEx+0x412a)[0x7ff020ac781a]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalCodeEx+0x88e)[0x7ff020acab7e]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalFrameEx+0x563a)[0x7ff020ac8d2a]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalCodeEx+0x88e)[0x7ff020acab7e]
/opt/Python2.7/lib/libpython2.7.so.1.0(+0x76f28)[0x7ff020a48f28]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyObject_Call+0x53)[0x7ff020a199a3]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalFrameEx+0x412a)[0x7ff020ac781a]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalCodeEx+0x88e)[0x7ff020acab7e]
/opt/Python2.7/lib/libpython2.7.so.1.0(+0x76f28)[0x7ff020a48f28]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyObject_Call+0x53)[0x7ff020a199a3]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalFrameEx+0x412a)[0x7ff020ac781a]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalCodeEx+0x88e)[0x7ff020acab7e]
/opt/Python2.7/lib/libpython2.7.so.1.0(+0x76f28)[0x7ff020a48f28]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyObject_Call+0x53)[0x7ff020a199a3]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalFrameEx+0x412a)[0x7ff020ac781a]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalCodeEx+0x88e)[0x7ff020acab7e]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalFrameEx+0x563a)[0x7ff020ac8d2a]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalCodeEx+0x88e)[0x7ff020acab7e]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalFrameEx+0x563a)[0x7ff020ac8d2a]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalCodeEx+0x88e)[0x7ff020acab7e]
/opt/Python2.7/lib/libpython2.7.so.1.0(+0x76f28)[0x7ff020a48f28]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyObject_Call+0x53)[0x7ff020a199a3]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalFrameEx+0x412a)[0x7ff020ac781a]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalCodeEx+0x88e)[0x7ff020acab7e]
/opt/Python2.7/lib/libpython2.7.so.1.0(+0x76f28)[0x7ff020a48f28]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyObject_Call+0x53)[0x7ff020a199a3]
.../p27/lib/python2.7/lib-dynload/_functools.so(+0x1764)[0x7ff01827a764]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyObject_Call+0x53)[0x7ff020a199a3]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalFrameEx+0x4ddd)[0x7ff020ac84cd]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalCodeEx+0x88e)[0x7ff020acab7e]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalFrameEx+0x563a)[0x7ff020ac8d2a]
```

```
*** glibc detected *** .../p27/bin/python: corrupted double-linked list: 0x0000000009e18c50 ***
======= Backtrace: =========
/lib64/libc.so.6(+0x76166)[0x7f6fb6e02166]
/lib64/libc.so.6(+0x79f9a)[0x7f6fb6e05f9a]
/lib64/libc.so.6(__libc_malloc+0x71)[0x7f6fb6e069a1]
.../p27/lib/python2.7/site-packages/matplotlib/_path.so(_ZN12_path_module14convert_to_svgERKN2Py5TupleE+0x92c)[0x7f6fa080887c]
.../p27/lib/python2.7/site-packages/matplotlib/_path.so(_ZN2Py15ExtensionModuleI12_path_moduleE21invoke_method_varargsEPvRKNS_5TupleE+0x18)[0x7f6fa0813828]
.../p27/lib/python2.7/site-packages/matplotlib/_path.so(method_varargs_call_handler+0x15d)[0x7f6fa082e20d]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalFrameEx+0x5910)[0x7f6fb7abf000]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalCodeEx+0x88e)[0x7f6fb7ac0b7e]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalFrameEx+0x563a)[0x7f6fb7abed2a]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalCodeEx+0x88e)[0x7f6fb7ac0b7e]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalFrameEx+0x563a)[0x7f6fb7abed2a]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalCodeEx+0x88e)[0x7f6fb7ac0b7e]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalFrameEx+0x563a)[0x7f6fb7abed2a]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalCodeEx+0x88e)[0x7f6fb7ac0b7e]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalFrameEx+0x563a)[0x7f6fb7abed2a]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalCodeEx+0x88e)[0x7f6fb7ac0b7e]
/opt/Python2.7/lib/libpython2.7.so.1.0(+0x76f28)[0x7f6fb7a3ef28]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyObject_Call+0x53)[0x7f6fb7a0f9a3]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalFrameEx+0x412a)[0x7f6fb7abd81a]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalCodeEx+0x88e)[0x7f6fb7ac0b7e]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalFrameEx+0x563a)[0x7f6fb7abed2a]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalCodeEx+0x88e)[0x7f6fb7ac0b7e]
/opt/Python2.7/lib/libpython2.7.so.1.0(+0x76f28)[0x7f6fb7a3ef28]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyObject_Call+0x53)[0x7f6fb7a0f9a3]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalFrameEx+0x412a)[0x7f6fb7abd81a]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalCodeEx+0x88e)[0x7f6fb7ac0b7e]
/opt/Python2.7/lib/libpython2.7.so.1.0(+0x76e21)[0x7f6fb7a3ee21]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyObject_Call+0x53)[0x7f6fb7a0f9a3]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalFrameEx+0x412a)[0x7f6fb7abd81a]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalCodeEx+0x88e)[0x7f6fb7ac0b7e]
/opt/Python2.7/lib/libpython2.7.so.1.0(+0x76f28)[0x7f6fb7a3ef28]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyObject_Call+0x53)[0x7f6fb7a0f9a3]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalFrameEx+0x412a)[0x7f6fb7abd81a]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalCodeEx+0x88e)[0x7f6fb7ac0b7e]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalFrameEx+0x563a)[0x7f6fb7abed2a]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalCodeEx+0x88e)[0x7f6fb7ac0b7e]
/opt/Python2.7/lib/libpython2.7.so.1.0(+0x76f28)[0x7f6fb7a3ef28]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyObject_Call+0x53)[0x7f6fb7a0f9a3]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalFrameEx+0x412a)[0x7f6fb7abd81a]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalCodeEx+0x88e)[0x7f6fb7ac0b7e]
/opt/Python2.7/lib/libpython2.7.so.1.0(+0x76f28)[0x7f6fb7a3ef28]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyObject_Call+0x53)[0x7f6fb7a0f9a3]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalFrameEx+0x412a)[0x7f6fb7abd81a]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalCodeEx+0x88e)[0x7f6fb7ac0b7e]
/opt/Python2.7/lib/libpython2.7.so.1.0(+0x76f28)[0x7f6fb7a3ef28]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyObject_Call+0x53)[0x7f6fb7a0f9a3]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalFrameEx+0x412a)[0x7f6fb7abd81a]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalCodeEx+0x88e)[0x7f6fb7ac0b7e]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalFrameEx+0x563a)[0x7f6fb7abed2a]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalCodeEx+0x88e)[0x7f6fb7ac0b7e]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalFrameEx+0x563a)[0x7f6fb7abed2a]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalCodeEx+0x88e)[0x7f6fb7ac0b7e]
/opt/Python2.7/lib/libpython2.7.so.1.0(+0x76f28)[0x7f6fb7a3ef28]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyObject_Call+0x53)[0x7f6fb7a0f9a3]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalFrameEx+0x412a)[0x7f6fb7abd81a]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalCodeEx+0x88e)[0x7f6fb7ac0b7e]
/opt/Python2.7/lib/libpython2.7.so.1.0(+0x76f28)[0x7f6fb7a3ef28]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyObject_Call+0x53)[0x7f6fb7a0f9a3]
.../p27/lib/python2.7/lib-dynload/_functools.so(+0x1764)[0x7f6faf270764]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyObject_Call+0x53)[0x7f6fb7a0f9a3]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalFrameEx+0x4ddd)[0x7f6fb7abe4cd]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalCodeEx+0x88e)[0x7f6fb7ac0b7e]
/opt/Python2.7/lib/libpython2.7.so.1.0(PyEval_EvalFrameEx+0x563a)[0x7f6fb7abed2a]
```

So, in summary: up until 760b2fc everything is fine, but with the following commit 7b39e78b, my program crashes randomly when drawing things with matplotlib, usually after a few minutes of drawing plots.

As the crashes appear randomly I cannot be absolutely sure that up until 760b2fc the problem really is not appearing, but as I let it run for an hour without crashes I'm pretty confident.

I hope you can find out what's going on, I'm certainly not an expert in these things. One guess of me would be that something writes over the edge of the string array, and this then sometimes happens to be a doubly linked list which gets corrupted.
